packages/grid/x-data-grid/src/hooks/features/rows/useGridRowsMeta.ts

```diff
@@ -50,34 +52,33 @@ export const useGridRowsMeta = (
   // Inspired by https://github.com/bvaughn/react-virtualized/blob/master/source/Grid/utils/CellSizeAndPositionManager.js
   const lastMeasuredRowIndex = React.useRef(-1);
   const hasRowWithAutoHeight = React.useRef(false);
-  const rowHeightFromDensity = useGridSelector(apiRef, gridDensityRowHeightSelector);
+  const densityFactor = useGridSelector(apiRef, gridDensityFactorSelector);
   const filterModel = useGridSelector(apiRef, gridFilterModelSelector);
   const paginationState = useGridSelector(apiRef, gridPaginationSelector);
   const sortModel = useGridSelector(apiRef, gridSortModelSelector);
   const currentPage = useGridVisibleRows(apiRef, props);
-
   const pinnedRows = useGridSelector(apiRef, gridPinnedRowsSelector);
 
+  const derivedRowHeight = React.useMemo(
```

Comment:
```suggestion
  const rowHeightFromDensity = React.useMemo(
```

To reduce the diff size?